### DISCUSSION TITLE
Can we set proxy for dendrite?

### DISCUSSION THREAD

**Discussion by Curious-r** [matrix-org/dendrite#3289] [Help & Support] [answered]
My server is in a constrained network environment, which is not able to connect to martrix.org. So I set an env:
`export ALL_PROXY="socks5://127.0.0.1:2080"`
Then I can curl to www.matrix.org smoothly, but my homeserver in this host still can't connect to matrix.org. Therefore I'd like to ask what the correct way to configure a proxy for Dendrite is.
Thanks!

**Comment by Curious-r** [comment, marked as answer]
Well, it seems that `ALL_PROXY` won't work, but `HTTP_PROXY` `HTTPS_PROXY` will.

**Reply by S7evinK** [reply]
Correct, Dendrite uses `ProxyFromEnvironment` provided by the [standard library](https://pkg.go.dev/net/http#ProxyFromEnvironment).
So you can use `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` to set the proxy.
